### DISCUSSION TITLE
Help with loader in next.config.js please!

### DISCUSSION THREAD

**Discussion by hatchli** [vercel/next.js#11575] [Help] [answered]
I cannot figure out why my loader is not working. I'm trying to render a pdf file via `react-pdf` using `url-loader`. I modified the next.config.js as the docs seemed to suggest, but no luck. Same result regardless.
I am getting :
```
./components/PDF/sample.pdf 1:0
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```

`next.config.js`
```
const withPlugins = require("next-compose-plugins");
const withTM = require("next-transpile-modules");
const withOptimizedImages = require("next-optimized-images");
const withFonts = require("next-fonts");
const withCSS = require("@zeit/next-css");
module.exports = {
  env: {
    STRIPE_PUBLIC_KEY: "your_stripe_public_key_here",
    API_URL: "http://localhost:3001"
  },
  webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
    // Fixes npm packages that depend on `fs` module
    if (!isServer) {
      config.node = {
        fs: "empty"
      };
    }
    config.module.rules.push({
      test: /\.pdf$/,
      use: {
        loader: "url-loader",
        options: {
          limit: 10000000,
          name: "[name].[ext]"
        }
      }
    }),
      config.resolve.modules.push(__dirname);

    config.resolve.alias = {
      ...config.resolve.alias
    };

    return config;
  }
};

module.exports = withPlugins(
  [
    [
      withTM,
      {
        transpileModules: ["reusecore", "common"]
      }
    ],
    [
      withOptimizedImages,
      {
        mozjpeg: {
          quality: 90
        },
        webp: {
          preset: "default",
          quality: 90
        }
      }
    ],
    withFonts,
    withCSS
  ],
  {
    distDir: "../../dist/functions/next"
  }
);

```

Next.js version 9.3.0

**Comment by coycoylaniba** [comment]
hey man, have you figure this out?

**Comment by vladimirevstratov** [comment, marked as answer]
For file loading from local resource you can use this. 

1. In your next.config.js in module.exports :

`output: {
    path: path.resolve(__dirname, 'static'),
  },`

2. And also add here:

```
webpack: (config, options) => {
    config.module.rules.push({
      test: /\.pdf/,
      type: 'asset/resource',
      generator: {
        filename: 'static/[hash][ext]',
      },
    })

    return config
},
```

3. And the last, in types.d.ts in root folder of project add:

`declare module '*.pdf'`

👏

After this steps you can use files like this:

`import PrivacyPolicy from '/public/Privacy_Policy_ready.pdf'`

And in jsx:

```
<a href={PrivacyPolicy} target="_blank" rel="noreferrer">
  policy
</a>
```

About custom webpack config: https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config
About asset management in webpack: https://webpack.js.org/guides/asset-management/
About output path configure in webpack: https://webpack.js.org/configuration/output/#outputpath

**Reply by moulie415** [reply]
thanks this worked for me!

**Reply by Fernando-Dev24** [reply]
2/2/2022
This work for me as well, thank you

**Reply by capscode** [reply]
i have added loader for my markdown file but after adding the loader, nextjs started giving error for other file types like .txt, .xml etc.
and giving error to add loader for those files as well.
and after adding loader for txt file or xml file its working well.

So my ask is do we need to add loaders for all the file type if we have loader for 1 ?

**Comment by Abdelmuttalib** [comment]
After applying the steps and trying to run the project with `yarn dev`, i get this error 


'error - Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error
ReferenceError: path is not defined
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:190:29)'
    

Although removing this line or step
`output: { path: path.resolve(__dirname, 'static'), },` 
from module.exports fixed the error.

**Reply by kair0span** [reply]
I run into the same issue. Could someone help with that?

**Reply by mwx27** [reply]
Maybe you need add:
const path = require('path')

**Reply by ZiaulKhan** [reply]
Same error...anybody found the solution for this...?
